### ISSUE TITLE
Review the VAOS calendar draft PR for accessibility

### ISSUE THREAD

**Issue by 1Copenut** [department-of-veterans-affairs/va.gov-team#2569]
WIP PR: https://github.com/department-of-veterans-affairs/vets-website/pull/10877

**Comment by 1Copenut** [comment]
@narin && @markgreenburg I took a look at this on Friday and have some general 508 comments and a Codepen for proof of concept: https://codepen.io/tpierce_402/pen/LYYxLxN

========================================
HTML
* Codepen with the [calendar mocked up using a table](https://codepen.io/tpierce_402/pen/LYYxLxN)
* `<i>` should have an `aria-hidden=“true”` attribute so it’s not a stop for virtual cursor
* Need a way to semantically represent one or both options (AM/PM) selected on a closed cell. 
* I’d like to include an `aria-controls` to the button. This will need unique IDs for each AM/PM “block” and the attribute should only appear on the button when its block is revealed. Otherwise we’re looking for IDs that don’t exist in the DOM.

========================================
AXE-CORE
* Checkboxes for AM/PM "block" need to be grouped in a `<fieldset> <legend>` group to avoid errors.
* The checkbox `<legend>` should have `sr-only` text or an `aria-describedby` attribute to explain what the AM/PM checkboxes represent. This text could read like "Do you prefer a morning (AM) or an afternoon (PM) appointment?"

========================================
ZOOM
* Layout becomes difficult to read at 300%, breaks down at 400%

========================================
KEYBOARD
* Where does focus go when I press Next or Previous? I think it should land on the next or previous month's heading or possibly the table `<caption>`. We'll have to experiment with the caption.
* Hit space on the checkbox labels is a bit small. Can we make each checkbox/radio "box" the hit space? Google recommends 8px ( https://developers.google.com/web/fundamentals/accessibility/accessible-styles ). 
* The checkboxes should be the next focusable elements after the toggle that opens them.

========================================
SCREENREADER (VO)
* The only way to navigate is forward, backward by one node. Can’t navigate in two dimensions like we can with a table. This is one of the big benefits of a table over a collection of cells.
* We can end run hearing the entire month, day, year, read out on every “cell” by using a table. I feel this will be a better aural experience for AT users.

**Comment by narin** [comment]
Thank you @1Copenut.  The item with the highest LOE would be converting the calendar to using a table instead of divs.  I'd be interested to know if using the table `roles` outlined [here](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Rowgroup_Role) would be an acceptable solution, either long term, or for the interim?

**Comment by 1Copenut** [comment]
@narin I'm testing ^^ out this morning and will follow up when I have the big three screen readers tested.

**Comment by 1Copenut** [comment]
@narin The MDN article you offered that applied roles to the divs for headings, rows, and cells tested well on assistive devices (VoiceOver, JAWS, NVDA). All three exhibited the same user flow as a standard table.

For Day 1, let's go ahead with adding the relevant `role` attributes to the existing code base and re-test ASAP. I can revise the findings and we can focus in on the 1A items for axe-core scans.

For posterity, the examples I tested were both simple Codepens I mocked up:
* [Divs with relevant table roles](https://codepen.io/tpierce_402/pen/pooeJGa)
* [Table-based calendar widget](https://codepen.io/tpierce_402/pen/LYYxLxN)

/cc @markgreenburg

**Comment by 1Copenut** [comment]
@andreahewitt-odd I'm going to move this one to Review. I've talked to Mark about the next steps and will start the full app audit in the next day or two.
